service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,227 @@
+            <parameter name="createOrderItem" type="Boolean" default="true"/>
+            <parameter name="packageInfoList" type="List">
+                <description>See get#AutoPackageInfo interface for details</description>
+                <parameter name="packageInfo" type="Map"/>
+            </parameter>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="shipment"/>
+        </out-parameters>
+        <actions>
+            <!--Todo: store value of optionValue in ShippingGatewayOption entity for SgoApiToken-->
+            <if condition="!packageInfoList">
+                <return message="No package info found, not getting UPS shipping rates"/>
+            </if>
+
+            <entity-find-one entity-name="mantle.order.OrderPart" value-field="orderPart"/>
+
+            <set field="fromPartyId" from="orderPart.vendorPartyId"/>
+            <if condition="!fromPartyId">
+                <return message="Vendor party not set on order part ${orderId}:${orderPartSeqId}"/>
+            </if>
+            <set field="toPartyId" from="orderPart.customerPartyId"/>
+            <if condition="!toPartyId">
+                <return message="To party not set on order part ${orderId}:${orderPartSeqId}"/>
+            </if>
+
+            <set field="carrierPartyId" from="carrierPartyId ?: orderPart.carrierPartyId"/>
+            <if condition="!carrierPartyId || carrierPartyId == '_NA_'">
+                <return message="Carrier not set on order part ${orderId}:${orderPartSeqId}"/>
+            </if>
+            <set field="shipmentMethodEnumId" from="shipmentMethodEnumId ?: orderPart.shipmentMethodEnumId"/>
+            <if condition="!shipmentMethodEnumId">
+                <return message="Shipment method not set on order part ${orderId}:${orderPartSeqId}"/>
+            </if>
+
+            <!--Get gatewayServiceCode of shipmentMethodEnumId from ShippingGatewayMethod-->
+            <entity-find-one entity-name="mantle.shipment.carrier.ShippingGatewayMethod" value-field="shippingGatewayMethod">
+                <field-map field-name="shippingGatewayConfigId"/>
+                <field-map field-name="carrierPartyId"/>
+                <field-map field-name="shipmentMethodEnumId"/>
+            </entity-find-one>
+            <if condition="shippingGatewayMethod">
+                <set field="serviceCode" from="shippingGatewayMethod.gatewayServiceCode"/>
+            </if>
+
+            <!-- Get description of shipmentMethodEnumId from CarrierShipmentMethod-->
+            <entity-find-one entity-name="mantle.shipment.carrier.CarrierShipmentMethod" value-field="carrierShipmentMethod">
+                <field-map field-name="shipmentMethodEnumId"/>
+                <field-map field-name="carrierPartyId"/>
+            </entity-find-one>
+            <if condition="carrierShipmentMethod">
+                <set field="description" from="carrierShipmentMethod.description"/>
```

Comment:
I think this is description of carrier shipment method and not of shipment method. Can you check this